File: CPP/ByTier/A/1852_A_Ntarsis_Set.cpp
// Problem: CF 1852 A - Ntarsis' Set
// https://codeforces.com/contest/1852/problem/A

/*
Algorithm: Ntarsis' Set
Techniques: Two-pointer technique, simulation with optimized index tracking

Time Complexity: O(n + k)
Space Complexity: O(n)

The problem simulates removing elements from a sorted set S = {1, 2, ..., 10^1000} based on the indices in array `a`.
Each day, we remove the a[1]-th, a[2]-th, ..., a[n]-th smallest numbers in S.
We are to find the smallest number remaining after k such days.

This implementation tracks how many elements have been removed via an iterative process using two pointers.
The key idea is to precompute the next valid element in the sequence as we simulate removals:
- `nw` represents the current pointer or offset in the sequence.
- `j` tracks how far we can go before we hit a point where a[j+1] - j <= nw.
- We increment `j` as long as it's safe to remove elements up to that point.
*/

#include <stdio.h>
#include <iosfwd>

using namespace std;
const int N = 2e5 + 5;
int T, n, k, a[N], s[N];

int main() {
  scanf("%d", &T);
  while (T--) {
    scanf("%d %d", &n, &k);
    for (int i = 1; i <= n; ++i)
      scanf("%d", &a[i]);
    long long nw = 1; // Starting position in the sequence
    for (int i = 1, j = 0; i <= k; ++i) {
      // Move j forward while it's safe to remove elements up to current value
      while (j < n && a[j + 1] - j <= nw)
        ++j;
      nw += j; // Update the next position based on how many were removed
    }
    printf("%lld\n", nw);
  }
}


// https://github.com/VaHiX/codeForces/

File: Python/ByRound/1852/1852_A_Ntarsis_Set.py
# Problem: CF 1852 A - Ntarsis' Set
# https://codeforces.com/contest/1852/problem/A

"""
Problem: Ntarsis' Set
Purpose: Given a set S initially containing integers 1, 2, ..., 10^1000 in sorted order,
         and an array 'a' of n distinct indices, each day we remove the a1-th, a2-th, ..., an-th smallest elements.
         After k days, find the smallest remaining element in S.

Algorithms/Techniques:
- Simulation with two pointers and incremental updates
- Greedy approach to determine which elements are skipped or removed

Time Complexity: O(n + k)
Space Complexity: O(1)

Approach:
- We simulate the process step by step.
- Start with ans = 1, and use increaser to track how many elements were skipped in previous steps.
- For each of the k days, update the answer by adding increaser (number of elements skipped).
- Use a pointer (curr_idx) to track which element from array 'a' is being processed next.
- If the current position in array 'a' can be handled (i.e., does not exceed the count of elements), increment the increaser and move the pointer.
- Otherwise, proceed with increasing the answer.

"""

import sys

input = sys.stdin.readline
for _ in range(int(input())):
    n, k = map(int, input().split())
    arr = tuple(map(int, input().split()))
    if arr[0] != 1:
        print(1)
        continue
    ans = 1
    increaser = 1
    curr_idx = 1
    while k > 0:
        # If index is valid and the current element can be skipped without going beyond what's available
        if (curr_idx) < n and (ans + increaser >= arr[curr_idx]):
            increaser += 1
            curr_idx += 1
            continue
        ans += increaser
        k -= 1
    print(ans)


# https://github.com/VaHiX/codeForces/